aten/src/ATen/function_wrapper.py

```diff
@@ -274,18 +282,164 @@ def __getitem__(self, x):
             return r
         return self.parent[x]
 
+if TYPE_HINTS:
+    Environment = TypedDict('Environment', {
```

Comment:
Some day, we'll actually have documentation for each of these fields :) (not this patch :)